core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ApplicationSummaryInfo.scala

```diff
@@ -47,7 +47,8 @@ case class ApplicationSummaryInfo(
     ioMetrics: Seq[IOAnalysisProfileResult],
     sysProps: Seq[RapidsPropertyProfileResult],
     sqlCleanedAlignedIds: Seq[SQLCleanAndAlignIdsProfileResult],
-    sparkRapidsBuildInfo: Seq[SparkRapidsBuildInfoEvent])
+    sparkRapidsBuildInfo: Seq[SparkRapidsBuildInfoEvent],
+    stageDiagnostics: Seq[StageDiagnosticResult])
```

Comment:
If this is just a supposed to be a view, why do we need to add that to the ApplicationSummaryInfo?
Again, this means that the table lives in memory and not limited to dumping the view.

Reply:
thanks @amahussein! could you elaborate a bit on this?

Reply:
I mean that we can generate those records for the sake of generating the report, and we do not have to store them in the ApplicationSummaryInfo.
ApplicationSummaryInfo fields are the ones that provide some information about the application and then it can be consumed by other modules inside scala (i.e., AutoTuner).